src/ts/components/playbacktimelabel.ts

```diff
@@ -132,7 +134,15 @@ export class PlaybackTimeLabel extends Label<PlaybackTimeLabelConfig> {
       }
     };
 
+    let playbackReadyHandler = () => {
+      // Set time format depending on source duration
+      this.timeFormat = Math.abs(player.isLive() ? player.getMaxTimeShift() : player.getDuration()) >= 3600 ?
+      StringUtils.FORMAT_HHMMSS : StringUtils.FORMAT_MMSS;
+      playbackTimeHandler();
+    };
+
     player.on(player.exports.PlayerEvent.TimeChanged, playbackTimeHandler);
+    player.on(player.exports.PlayerEvent.Ready, playbackReadyHandler);
```

Comment:
The only problem that I'm seeing is that imho duration should be available after `SourceLoaded`. If that's not the case, it could be a bug on the Player SDK (I guess Android in this case). If duration changes afterwards, there should be a `DurationChanged` event. If this also does not happen on Android, maybe we should argue that they should introduce such an event, since it seems to be relevant.

If we still decide to treat this as a UI issue, I'm not convinced setting the duration after the `Ready` event is the best approch. `Ready` indicates that content is playable, but duration should be already available much earlier.

Reply:
@matamegger @strangesource are you guys sure all this is handled correctly on the Android Player side? Meaning that everything is by design? Are we sure this should be addressed on the UI side?

@stonko1994 do you mind sharing your opinion as well?

Thanks.

Reply:
in fact, I can't even play this in the Chromecast receiver, I tried the SDK version 3. Also note that in the jira item the version number is 8.80, this would be a Web Player... 

at the same time, in the jira incident to use the Android samples...

Reply:
but even without to test it, it's clear that the "mm:ss" could be selected before the duration is known, so if we know what event to await we can merge this and close the item.

If we don't know in what setup the client used, the manual test can be done later once we have this information.

Reply:
Hi @bitsmith1010, @wasp898 👋,

as already [stated in the Android issue](https://bitmovin.atlassian.net/browse/PA-1661?focusedCommentId=235624): When we start casting a stream, we have to wait for the duration to be transmitted from the cast receiver. This takes some time and can result in an invalid duration being available when the Bitmovin Web UI starts up. 

I see your point that the duration should be available after the loaded event. Unfortunately this is currently not the case when the cast device is already connected before loading the source into the player as the loaded event is solely emitted for local source loading. 

Having said that, I do see a discrepancy on the UI side here where the **duration is updated but the format is not** resulting in the "cropped" duration. 
This caused my reasoning that the issue could/should be handled in the UI.

Reply:
I would agree with both sides here 🤔!

As I understand it, there seems to be an issue on the Android SDK side where the duration is not available where it should be, and the change of it is not indicated via a `DurationChanged` event.

On the other hand, I also agree that if the UI already updates the duration on other occasions (in this case, based on `TimeChanged` events), why not also update the time format?

Reply:
Okay @strangesource, thanks.

Then my only suggestion to @bitsmith1010 is that we update the format both on initialization (as it was) and after `Ready` event, and not only on `Ready`. The suggestion from David is also valid.